### PR TITLE
kinflate: improved README and added getting-started

### PR DESCRIPTION
**Pull request by droot**
Improves README.md
Added Getting Started guide

### CODE REVIEW

File: cmd/kinflate/README.md [hunk 1]
```diff
@@ -1,12 +1,8 @@
...
-`kinflate` is a kubernetes cluster configuration
-utility.
-
-It reads a cluster app definition from a
-directory of YAML files, and creates new YAML suitable
-for directly piping to `kubectl apply`.
+# Overview
 
+`kinflate` is a command line interface to manage Kubernetes Application 
+configuration.  It reads an app definition from a directory of YAML files, and
```
Comment (ahmetb) [inline cmd/kinflate/README.md]: What is an  "app definition"? Is it Deployment/ReplicaSet/...? Does this not work with services?

Change YAML files to "YAML manifest files".

File: cmd/kinflate/README.md [hunk 2]
```diff
@@ -1,12 +1,8 @@
...
-It reads a cluster app definition from a
-directory of YAML files, and creates new YAML suitable
-for directly piping to `kubectl apply`.
+# Overview
 
+`kinflate` is a command line interface to manage Kubernetes Application 
+configuration.  It reads an app definition from a directory of YAML files, and 
+creates new YAML suitable for directly piping to `kubectl apply`.
 
 ## Declarative Application Management
```
Comment (ahmetb) [inline cmd/kinflate/README.md]: this paragraph is not useful to anyone, do not expect users to read the proposal or even get familiar with the acronym. Kubernetes users already know that YAML files are declarative application management.

File: cmd/kinflate/README.md [hunk 3]
```diff
@@ -25,14 +21,79 @@ control commits.  It encourages forking a configuration,
...
-go get k8s.io/kubectl/cmd/kinflate
+
+After running the above command, `kinflate` should get installed in your `GOPATH/bin` directory.
+
+## Usage
+
+This section explains sub-commands of `kinflate` using examples. 
+
+### init 
+`kinflate init` initializes the Kubernetes Application by creating an
```
Comment (ahmetb) [inline cmd/kinflate/README.md]: "initializes the Kubernetes Application" doesn't make a lot of sense. Maybe like this, but I'm not sure if it's accurate:

> `kinflate init` initializes a directory with a sample Deployment manifest file

File: cmd/kinflate/README.md [hunk 4]
```diff
@@ -25,14 +21,79 @@ control commits.  It encourages forking a configuration,
...
+```shell
+kinflate init
 ```
 
+### set nameprefix
+
+`kinflate set nameprefix` updates the nameprefix field in the [manifest].
+`kinflate inflate` uses the prefix to generate prefixed names for Kubernetes resources.
+
+#### examples
```
Comment (ahmetb) [inline cmd/kinflate/README.md]: These don't have to be titles, you can just say "Examples:"

File: cmd/kinflate/README.md [hunk 5]
```diff
@@ -25,14 +21,79 @@ control commits.  It encourages forking a configuration,
...
+
+# generate application configuration and apply it to a Kubernetes cluster
+kinflate inflate -f <path/to/application/dir> | kubectl apply -f
+```
+
+
 ## Demos and tutorial
 
+ * [Getting Started](getting_started.md)
  * [Short demo](shortDemo.md)
```
Comment (ahmetb) [inline cmd/kinflate/README.md]: "short demo", "long demo" titles are not useful. The type of example titles I'd like to see:

- Using kinflate to inject environment variable
- Using kinflate to modify annotation
- Using kinflate to declare ports
- Using kinflate to [...]

File: cmd/kinflate/getting_started.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+If you have `kinflate` installed already, then you can skip this step. `kinflate` can be installed using `go get`:
+
+```shell
+go get -u k8s.io/kubectl/cmd/kinflate
+```
+This fetches kinflate and install `kinflate` executable under `GOPATH/bin` - you will want that on your $PATH.
+
+
+### Get off-the-shelf MySQL configs for Kubernetes
+First step is to get basic MySQL configs for Kubernetes in place. Lets get them from the internet.
```
Comment (ahmetb) [inline cmd/kinflate/getting_started.md]: replace this whole sentence with:

> Download a sample MySQL deployment manifest file:

File: cmd/kinflate/README.md [hunk 7]
```diff
@@ -25,14 +21,79 @@ control commits.  It encourages forking a configuration,
...
+`kinflate inflate` generates application configuration by applying
+transformations (e.g. name-prefix, label, annotations) specified in the [manifest].
+
+#### examples
+```shell
+# generate application configuration and output to stdout
+kinflate inflate -f <path/to/application/dir>
+
+# generate application configuration and apply it to a Kubernetes cluster
+kinflate inflate -f <path/to/application/dir> | kubectl apply -f
```
Comment (ahmetb) [inline cmd/kinflate/README.md]: you need to make this `kubectl apply -f-` otherwise it won't work I believe.

Comment (droot) [reply cmd/kinflate/README.md]: Good catch. Thanks

File: cmd/kinflate/README.md (hunk 4 above)
Comment (droot) [reply cmd/kinflate/README.md]: Done.

File: cmd/kinflate/README.md (hunk 3 above)
Comment (droot) [reply cmd/kinflate/README.md]: Fixed.

File: cmd/kinflate/getting_started.md (hunk 6 above)
Comment (droot) [reply cmd/kinflate/getting_started.md]: Fixed :)

File: cmd/kinflate/README.md (hunk 5 above)
Comment (droot) [reply cmd/kinflate/README.md]: Great point. Thats more aligned with what users would be looking for. Added a note, will be adding in next set of changes.

**Pull request merged**